wrapper/src/main/java/software/amazon/jdbc/ConnectionProvider.java

```diff
@@ -49,4 +62,9 @@ Connection connect(
    */
   Connection connect(@NonNull String url, @NonNull Properties props)
       throws SQLException; // TODO: do we need this method?
+
+  /**
+   * Release any resources used by this ConnectionProvider
+   */
+  void releaseResources();
```

Comment:
I would not extend interface with this method but allow each particular class to implement CanReleaseResources if needed.